recipes/paho-mqtt-cpp/all/conanfile.py

```diff
@@ -1,102 +1,120 @@
+        tc.variables["PAHO_BUILD_SAMPLES"] = False
+        tc.variables["PAHO_BUILD_STATIC"] = not self.options.shared
+        tc.variables["PAHO_BUILD_SHARED"] = self.options.shared
+        tc.variables["PAHO_WITH_SSL"] = self.options.ssl
+        tc.generate()
+        deps = CMakeDeps(self)
+        deps.generate()
+        vbuildenv = VirtualBuildEnv(self)
+        vbuildenv.generate(scope="build")
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
 
     def build(self):
-        # See this conversation https://github.com/conan-io/conan-center-index/pull/4096#discussion_r556119143
-        # Changed by https://github.com/eclipse/paho.mqtt.c/commit/f875768984574fede6065c8ede0a7eac890a6e09
-        # and https://github.com/eclipse/paho.mqtt.c/commit/c116b725fff631180414a6e99701977715a4a690
-        # FIXME: after https://github.com/conan-io/conan/pull/8053#pullrequestreview-541120387
-        if tools.Version(self.version) < "1.2.0" and tools.Version(self.deps_cpp_info["paho-mqtt-c"].version) >= "1.3.2":
-            raise ConanInvalidConfiguration("{}/{} requires paho-mqtt-c =< 1.3.1".format(self.name, self.version))
-
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        self._patch_sources()
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy("edl-v10", src=self._source_subfolder, dst="licenses")
-        self.copy("epl-v10", src=self._source_subfolder, dst="licenses")
-        self.copy("notice.html", src=self._source_subfolder, dst="licenses")
-        cmake = self._configure_cmake()
+        copy(self, "edl-v10", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "epl-v10", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        copy(self, "notice.html", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
         self.cpp_info.names["cmake_find_package"] = "PahoMqttCpp"
         self.cpp_info.names["cmake_find_package_multi"] = "PahoMqttCpp"
+        self.cpp_info.set_property("cmake_file_name", "PahoMqttCpp")
+
         target = "paho-mqttpp3" if self.options.shared else "paho-mqttpp3-static"
+        self.cpp_info.components["paho-mqttpp"].set_property("cmake_target_name", f"PahoMqttCpp::{target}")
         self.cpp_info.components["paho-mqttpp"].names["cmake_find_package"] = target
         self.cpp_info.components["paho-mqttpp"].names["cmake_find_package_multi"] = target
+        self.cpp_info.components["paho-mqttpp"].set_property("cmake_file_name", target)
```

Comment:
```suggestion
        self.cpp_info.components["paho-mqttpp"].set_property("cmake_file_name", f"PahoMqttCpp::{target}")
```

Reply:
this is cmake_file_name, not cmake_target_name - this is to match behaviour of the conan1 recipe which seems to use different configuration file names depending on if it's static or not.

Reply:
so it doesn't make sense, cmake_file_name property for components is ignored.

Reply:
I've edited suggested change